### PR TITLE
feat: add ecMul precompile

### LINKED ISSUE #363: feat: Explore / propose strategy (if applicable, MVP implementation) for EVM Precompiled contract -> 0x07 ecMul
# Feature Request

**Describe the Feature Request**

feat: Explore / propose strategy (if applicable, MVP implementation) for EVM Precompiled contract -> 0x07 ecMul

https://www.evm.codes/precompiled

### PR DESCRIPTION
**Pull request by pedrobergamini**
Time spent on this PR:
1 day

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

There is no ecMul implementation.
Resolves #363 

## What is the new behavior?

- Implements ecMul precompile and required tests.

### CODE REVIEW

File: src/utils/alt_bn128/alt_bn128_field.cairo [hunk 1]
```diff
@@ -0,0 +1,510 @@
+// SPDX-License-Identifier: MIT
```
Comment (ClementWalter) [inline src/utils/alt_bn128/alt_bn128_field.cairo:1]: is this copied from elsewhere?

File: src/utils/alt_bn128/bigint.cairo [hunk 2]
```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: MIT
```
Comment (ClementWalter) [inline src/utils/alt_bn128/bigint.cairo:1]: is this copied from elsewhere?

File: tests/integration/bytecode/test_cases.py [hunk 3]
```diff
@@ -1628,4 +1628,41 @@
         "id": "Precompiles - RIPEMD160 - playground test case",
         "marks": [pytest.mark.RIPEMD160, pytest.mark.Precompiles],
     },
+    {
+        "params": {
+            "value": 0,
+            "code": "60016000526002602052600160405260026060526040608060806000600663fffffffffa5060a051608051",
+            "calldata": "",
+            "stack": f"{str(0x15ED738C0E0A7C92E7845F96B2AE9C0A68A6A449E3538FC7FF3EBF7A5A18A2C4)},{str(0x30644E72E131A029B85045B68181585D97816A916871CA8D3C208C16D87CFD3)}",
```
Comment (ClementWalter) [inline tests/integration/bytecode/test_cases.py:1636]: `str` is redundant

File: tests/unit/src/kakarot/precompiles/test_ecadd.cairo [hunk 4]
```diff
@@ -29,30 +31,119 @@ func test__ecadd_impl{
...
+        array_0=bytes_expected_result,
+        array_1_len=output_len,
+        array_1=output,
+    );
+
+    return ();
+}
+
+@external
+func test__ecadd_via_staticcall{
```
Comment (ClementWalter) [inline tests/unit/src/kakarot/precompiles/test_ecadd.cairo:69]: do you miss a rebase?

**Review by ClementWalter — CHANGES_REQUESTED** [review]
I think that you started for you EC_ADD branch and need to rebase on main now

File: src/utils/alt_bn128/bigint.cairo (hunk 2 above)
Comment (pedrobergamini) [reply src/utils/alt_bn128/bigint.cairo]: It's the [alt_bn128](https://github.com/tekkac/cairo-alt_bn128) lib from tekkac, I had the reference in the main file but just added to all files

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**